x/ecocredit/server/core/query_projects_by_reference_id.go

```diff
@@ -0,0 +1,65 @@
+package core
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/types/ormutil"
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+// ProjectsByReferenceId queries projects by reference id.
+func (k Keeper) ProjectsByReferenceId(ctx context.Context, req *core.QueryProjectsByReferenceIdRequest) (*core.QueryProjectsByReferenceIdResponse, error) {
+	if req.ReferenceId == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "reference-id is empty")
```

Comment:
I like the use of status and codes here although its not consistent with other query implementations. I think this would be an improvement and we should probably open an issue to track such changes elsewhere.

Reply:
Actually, if someone is using the golang API directly, this would not be relevant. I think we should be using the sdk errors and invalid request here.

Reply:
It looks like we are using in basket queries. Also looks like this format is being used in the sdk modules.

Reply:
Yes. We are using this format in sdk